### PR TITLE
Made some changes, added some tests

### PR DESCRIPTION
**Pull request by HansHammel**
note: reverted the tab indention

nice work,. regards.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,28 +16,49 @@ In order to use Fixme all you need to do is install it:
...
-  line_length_limit:    1000
+  line_length_limit:    1000,
+  color:                true
+};
+```
+
+when called like so:
+
+```javascript
+fixme(options, function(outptu){
```
Comment (JohnPostlethwait) [inline README.md]: Looks like a typo.

File: bin/fixme.js [hunk 2]
```diff
@@ -1,356 +1,386 @@
...
+
+    var byline = require('byline'),
+        chalk = require('chalk'),
+        eventStream = require('event-stream'),
+        fs = require('fs'),
+        os = require('os'),
+        isBinaryFile = require('isbinaryfile'),
+        minimatch = require('minimatch'),
+        readdirp = require('readdirp'),
+//TODO: get rid of this dirty hack (removing the color codes aterwards)
```
Comment (JohnPostlethwait) [inline bin/fixme.js]: Another typo "aterwards"

File: .gitignore [hunk 3]
```diff
@@ -1,28 +1,44 @@
...
+# Dependency directories
+node_modules/
+jspm_packages/
+bower_components/
+
+# Optional npm cache directory
+.npm/
+
+# Optional REPL history
+.node_repl_history
```
Comment (JohnPostlethwait) [inline .gitignore]: No newline at the end. (git can have a hard time with this.)

File: README.md [hunk 4]
```diff
@@ -16,28 +16,49 @@ In order to use Fixme all you need to do is install it:
...
+When using it **programtically**
+
+_like so:_
 
 ```javascript
 var fixme = require('fixme');
 ```
 
-And finally; configure it when you call it:
+the follwing options can be configured:
```
Comment (JohnPostlethwait) [inline README.md]: typo "follwing"

File: bin/fixme.js [hunk 5]
```diff
@@ -1,356 +1,386 @@
...
+                else callback(stripAnsi(out));
+            }
+        });
+    };
+
+    if (!module.parent)
+        parseUserOptionsAndScan();
+    else
+        module.exports = parseUserOptionsAndScan;
+})();
```
Comment (JohnPostlethwait) [inline bin/fixme.js]: No newline

File: example/fixme.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+var fixme = require('../index.js');
+//var fixme = require('fixme');
```
Comment (JohnPostlethwait) [inline example/fixme.js]: Should this be deleted?

File: example/fixme.md.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+var fixme = require('../index.js');
+var path = require('path');
+//var fixme = require('fixme');
```
Comment (JohnPostlethwait) [inline example/fixme.md.js]: Should be removed?

File: example/fixme.md.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    file_encoding: 'utf8',
+    line_length_limit: 40000,
+    color: true,
+    md: true
+}, function (out) {
+    console.log(out);
+    //console.log(convert.toHtml(out));
+    fs.writeFileSync(path.join(__dirname,'out.md'),convert.toHtml(out),'utf8');
+    fs.writeFileSync(path.join(__dirname,'out.raw.md'),out,'utf8');
+});
```
Comment (JohnPostlethwait) [inline example/fixme.md.js]: Newline

File: test/fixme.test.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+            var filecontent = fs.readFileSync(path.join(__dirname, 'fixme.withcolor.' + (os.EOL === "\r\n" ? 'crlf' : 'lf') + '.txt'), 'utf8');
+            //reset console before assertion, so mocha can print
+            delete console.log;
+            expect(out.trim()).to.equal(filecontent.trim());
+            done();
+        });
+    });
+
+
+});
```
Comment (JohnPostlethwait) [inline test/fixme.test.js]: newline

**Comment by JohnPostlethwait** [comment]
Overall it looks great! Thanks for the PR! 

If you wouldn't mind rebasing it down to a single commit, fixing the few typos, unnecessary commented out code, and newlines and I will merge it.

Also, I see mocha tests in there, but I couldn't get them to run:

```bash
npm test --verbose
npm info it worked if it ends with ok
npm verb cli [ '/Users/johnpostlethwait/.nvm/versions/node/v4.2.6/bin/node',
npm verb cli   '/Users/johnpostlethwait/.nvm/versions/node/v4.2.6/bin/npm',
npm verb cli   'test',
npm verb cli   '--verbose' ]
npm info using npm@2.14.12
npm info using node@v4.2.6
npm verb run-script [ 'pretest', 'test', 'posttest' ]
npm info pretest fixme@0.3.0
npm info test fixme@0.3.0

> fixme@0.3.0 test /Users/johnpostlethwait/Documents/workspace/fixme
> mocha --reporter spec test/fixme.test.js
  fixme
npm verb unsafe-perm in lifecycle true
npm info fixme@0.3.0 Failed to exec test script
npm ERR! Test failed.  See above for more details.
npm verb exit [ 1, true ]
```

Any ideas on that?

**Comment by HansHammel** [comment]
Sorry for the late response. Currently I have very few time to work on this. If you find typos, please feel free to correct them, as I am no native speaker. And to be honest, I didn't get the point of you first comment, where is the typo?

The tests are run with `npm test`, using a **local** mocha.

There are currently no tests included for Linux, but you can generate them by uncommenting the line 375, 376 (bin/fixme.js) to write the reference files for the linux tests. To do so, uncomment the mentioned lines, run `npm test` and find the files in the projects root folder. Review them manually! Then, move them to the test directory, uncomment the lines 375, 376 and run `npm test` again. It should now pass. I did a Linux test on debian with latest node 4 weeks ago, so I am quite sure it works. note: meanwhile I made some more changes in my repo, e.g. removing the clockwise arrow because it wasn't supported by the IDE's font. 

The current commit (3/7/2016) is working on windows. And, yes, the windows tests were broken, too, but I don't know why. I am using another machine for this quick test, right now, so there may be further issues related to platform and/or node versions. Btw: One more thing to mention is, that in the windows console none of the default fonts will display the symbols, very annoying. 

Regards.

File: README.md (hunk 1 above)
Comment (JohnPostlethwait) [inline README.md]: The variable passed into the function here is "outptu" (I believe you meant "output")

**Pull request closed without merging**